site/content/docs/11-deploying.md

```diff
@@ -61,4 +61,100 @@ When deploying to [Now][], you can pass the environment variable into Now itself
 now -e SAPPER_TIMESTAMP=$(date +%s%3N)
 ```
 
-[Now]: https://zeit.co/now
\ No newline at end of file
+[Now]: https://zeit.co/now
+
+
+## Deploying to App Engine
+
+[Google App Engine](https://cloud.google.com/appengine/) is a fully managed serverless platform. The app engine will run the dynamic server so its not necessary to perform `npm run export`. Before deploying you need to add an `app.yaml` file to your project, that specifies the required node runtime:
+
+```
+# app.yaml
+runtime: nodejs10
+```
+
+A [custom build step](https://cloud.google.com/appengine/docs/standard/nodejs/running-custom-build-step) can be run by adding `"gcp-build": "run-p build"` as a `scripts` entry to your `package.json` scripts. This ensures that the build step will be performed before the application starts. Now you can deploy using the `gcloud app deploy` command.
+
+
+## Deploying to Docker
+
+[Docker](https://docs.docker.com/install/) can be used to build an image and run it on your local system. Doing this locally allows testing the build process before building and deploying it on a cloud platform. The Docker image can be build by adding a `Dockerfile` into your project:
+
+```Dockerfile
+# Dockerfile
+FROM mhart/alpine-node:12 as build
+
+WORKDIR /app
+COPY . .
+RUN npm ci
+RUN npm run build
+
+FROM mhart/alpine-node:12 as prod
```

Comment:
This step can be skipped and the build can use [`npm prune --prod`](https://github.com/sanderhahn/cloud-run-button-dockerfile/blob/master/Dockerfile#L6) if the polka dependency is updated in the template.

Reply:
If you mean doing `npm prune` instead of starting a starting from a new base image and doing `COPY --from`, that's not a good idea. While the files may not be present in the final image, they will still be present in one of its intermediate layers, and will contribute to its size.

Reply:
Oh, do you mean still doing the separate `COPY --from` below?

Reply:
Yes, was thinking it would start out with the fresh `mhart/alpine-node:slim-12`, at least the image size didn't increase.